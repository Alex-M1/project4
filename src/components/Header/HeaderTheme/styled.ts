import styled from 'styled-components';
import { IStyled } from 'common/types/styledTypes';

export const StThemeToggle = styled.button < IStyled > `
  position: absolute;
  top: 25px;
  right: 10%;
  width: 40px;
  height: 40px;
  border-radius: 50%;
  border: none;
  background: ${({ colors, theme }) => colors[theme].iconTheme};
  overflow: none;
  z-index: 1;
  &::after {
    content: '';
    position: absolute;
    top: 0;
    left: -10px;
    width: 40px;
    height: 40px;
    border-radius: 50%;
    border: none;
    background: ${({ colors, theme }) => colors[theme].iconThemeMoon};
  }
  &::before {
    content: '';
    position: absolute;
    top: 0;
    left: 0;
    width: 40px;
    height: 40px;
    border-radius: 50%;
    border: none;
    background: ${({ colors, theme }) => colors[theme].iconThemeSun};
    box-shadow: 0 0 20px ${({ colors, theme }) => colors[theme].iconThemeSun};
  }
`;
