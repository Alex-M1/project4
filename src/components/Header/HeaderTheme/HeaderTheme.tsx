import React from 'react';
import { StThemeToggle } from './styled';
import { useTheme } from '../../hooks/useTheme';

const HeaderTheme: React.FC = () => {
  const { colors, theme, changeTheme } = useTheme();
  return (
    <StThemeToggle
      theme={theme}
      colors={colors}
      onClick={changeTheme}
    />
  );
};
export default HeaderTheme;
