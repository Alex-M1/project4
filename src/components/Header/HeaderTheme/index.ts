import HeaderTheme from './HeaderTheme';

export default HeaderTheme;
