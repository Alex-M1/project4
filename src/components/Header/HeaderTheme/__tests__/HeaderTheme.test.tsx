import { mount, shallow } from 'enzyme';
import React from 'react';
import HeaderTheme from '../HeaderTheme';

jest.mock('react-i18next', () => ({
  useTranslation: jest.fn().mockImplementation(() => ({
    i18n: {
      changeLanguage: jest.fn(),
    },
  })),
}),
);
jest.mock('src/components/hooks/useTheme', () => ({
  useTheme: jest.fn().mockImplementation(() => ({
    colors: {
      dark: {},
      light: {},
    },
    theme: 'dark',
    changeTheme: jest.fn(),
  })),
}),
);

describe('HeaderTheme', () => {
  it('Should match snapshot', () => {
    const component = shallow(<HeaderTheme />);
    expect(component.html()).toMatchSnapshot();
  });
  it('should render StThemeToggle', () => {
    const component = mount(<HeaderTheme />);
    expect(component.find('styled__StThemeToggle')).toHaveLength(1);
  });
});
