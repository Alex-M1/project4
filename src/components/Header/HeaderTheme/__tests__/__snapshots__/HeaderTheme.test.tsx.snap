// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderTheme Should match snapshot 1`] = `"<button class=\\"styled__StThemeToggle-jPlXPr JBFkI\\"></button>"`;
